test/postgresql.autoupdate.test.js

```diff
@@ -677,4 +679,274 @@ describe('autoupdate', function() {
+            ds.define(noIndexSchema.name, noIndexSchema.properties, noIndexSchema.options);
+            ds.autoupdate([schema.name], cb);
+          },
+          cb => {
+            // Validate post-update indexes contain expected values
+            ds.connector.discoverModelIndexes(schema.name, (err, indexes) => {
+              if (err) {
+                return cb(err);
+              }
+
+              assert.deepEqual(indexes, {
+                Person_pkey: {
+                  table: 'Person',
+                  type: 'btree',
+                  primary: true,
+                  unique: true,
+                  keys: ['id'],
+                  order: ['ASC']},
+              });
+
+              cb();
+            });
+          },
+        ], (err) => {
+          assert(!err, err);
+          done();
+        });
+      });
+
+      describe('and index is modified', () => {
+        it('updates successfully', (done) => {
+          async.series([
+            cb => {
+              // Update schema
+              ds.define(schema.name, changedSchema.properties, changedSchema.options);
+              ds.autoupdate([schema.name], cb);
+            },
+            cb => {
+              // Validate post-update indexes contain expected values
+              ds.connector.discoverModelIndexes(schema.name, (err, indexes) => {
+                if (err) {
+                  return cb(err);
+                }
+
+                assert.deepEqual(indexes, {
+                  Person_pkey: {
+                    table: 'Person',
+                    type: 'btree',
+                    primary: true,
+                    unique: true,
+                    keys: ['id'],
+                    order: ['ASC']},
+                  uniqueNameIndex: {
+                    table: 'Person',
+                    type: 'btree',
+                    primary: false,
+                    unique: true,
+                    keys: ['firstname', 'lastname'],
+                    order: ['DESC', 'DESC']},
+                });
+
+                cb();
+              });
+            },
+          ], (err) => {
+            assert(!err, err);
+            done();
+          });
+        });
+
+        describe('for index uniqueness', () => {
+          const updatedSchema = {
+            ...schema,
+            options: {
+              ...schema.options,
+              indexes: {
+                uniqueNameIndex: {
+                  keys: {'firstName': 1, 'lastName': 1},
+                  options: {unique: true},
+                },
+              },
+            },
+          };
+
+          it('drops the existing index', async () => {
+            ds.define(updatedSchema.name, updatedSchema.properties, updatedSchema.options);
+            await ds.autoupdate([updatedSchema.name]);
+
+            const calls = spy.getCalls();
+            const match = calls.some(call => DROP_INDEX_REGEX.test(call.firstArg));
+            assert(match);
```

Comment:
I think it'd be better to add ` ds.connector.discoverModelIndexes(schema.name...)` so that we can see the result of this update.

Reply:
@agnes512 I can certainly add the validation against the `ds.connector.discoverModelIndexes()` result (and to the tests referenced below, as well.) Just to be clear, are you suggesting removing the check for the `DROP` SQL calls and replacing with _just_ the check against the modified schema? That should be fine since if the schema was changed, then the `DROP` must have happened.